pallets/pool-system/src/tranches.rs

```diff
@@ -1392,14 +1456,2731 @@ where
+			);
+		}
+
+		#[test]
+		fn next_id_works() {
+			let mut tranches = default_tranches();
+			let next = tranches.next_id();
+			let next_again = tranches.next_id();
+
+			assert_ne!(next, next_again)
+		}
+
+		struct TokenNameLen;
+		impl Get<u32> for TokenNameLen {
+			fn get() -> u32 {
+				16u32
+			}
+		}
+
+		struct TokenSymLen;
+		impl Get<u32> for TokenSymLen {
+			fn get() -> u32 {
+				8u32
+			}
+		}
+
+		// Replace should work if interest is lower than tranche w/ lower index ("next").
+		#[test]
+		fn replace_tranche_less_interest_than_next_works() {
+			let mut tranches = default_tranches();
+
+			// ensure we have an interest rate lower than the the left side tranche with a lower index, e.g. lower than 10% at index 1
+			let int_per_sec = Rate::one() / Rate::saturating_from_integer(SECS_PER_YEAR);
+			let min_risk_buffer = Perquintill::from_rational(4u64, 5);
+			let input = TrancheInput {
+				// setting to easily testable value for tranche replacement
+				seniority: Some(5),
+				tranche_type: TrancheType::NonResidual {
+					interest_rate_per_sec: int_per_sec,
+					min_risk_buffer: min_risk_buffer,
+				},
+				metadata: TrancheMetadata {
+					token_name: BoundedVec::<u8, TokenNameLen>::default(),
+					token_symbol: BoundedVec::<u8, TokenSymLen>::default(),
+				},
+			};
+
+			// verify replace tranche works with interest less than prev tranche as expected
+			// replacing last tranche
+			let replace_res = tranches.replace(2, input, SECS_PER_YEAR);
+			assert!(replace_res.is_ok());
+			assert_eq!(
+				tranches
+					.get_tranche(TrancheLoc::Index(2))
+					.unwrap()
+					.seniority,
+				5
+			);
+		}
+
+		// Replace must not work if new interest rate is greater than tranche w/ lower index ("next").
+		#[test]
+		fn replace_tranche_more_interest_than_next_throws() {
+			let mut tranches = default_tranches();
+			let min_risk_buffer = Perquintill::from_rational(4u64, 5);
+
+			// ensure we have an interest rate larger than the a tranche with a lower index, e.g. 10%
+			let int_per_sec = Rate::saturating_from_rational(11, 100)
+				/ Rate::saturating_from_integer(SECS_PER_YEAR)
+				+ One::one();
+			let input = TrancheInput {
+				// setting to easily testable value for tranche replacement, should not be changed from 0
+				seniority: Some(5),
+				tranche_type: TrancheType::NonResidual {
+					interest_rate_per_sec: int_per_sec,
+					min_risk_buffer: min_risk_buffer,
+				},
+				metadata: TrancheMetadata {
+					token_name: BoundedVec::<u8, TokenNameLen>::default(),
+					token_symbol: BoundedVec::<u8, TokenSymLen>::default(),
+				},
+			};
+
+			let replace_res = tranches.replace(2, input, SECS_PER_YEAR);
+			assert_eq!(
+				replace_res,
+				Err(DispatchError::Other(
+					"Invalid next tranche type. This should be catched somewhere else."
+				))
+			);
+			// verify unchanged from default val
+			assert_eq!(
+				tranches
+					.get_tranche(TrancheLoc::Index(2))
+					.unwrap()
+					.seniority,
+				0
+			);
```

Comment:
I would rather test for equality of the whole tranche here, than the equality of the `seniority`

Reply:
Apart from the `seniority` and the `tranche_type` it does not make sense to compare more since [the rest is boilerplate via `create_tranche`](https://github.com/centrifuge/centrifuge-chain/blob/tests%2Fcontinue-tranches-tests/pallets/pool-system/src/tranches.rs#L558):

Fixed in 9c5baf6e30868bfc6a011e68b09a866c233bacad